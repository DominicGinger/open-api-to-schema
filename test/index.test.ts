import openApiSchema, { Config, Required } from '../src/index'

describe('index', () => {
  test('Converts schema with optional fields', () => {
    const config: Config = {
      required: Required.ALL,
      optionalFields: {
        Pet: [ 'id' ]
      }
    }

    const schema = openApiSchema('./test/fixtures/petstore-expanded.yaml', config)
    expect(schema).toMatchSnapshot()
  })
})
