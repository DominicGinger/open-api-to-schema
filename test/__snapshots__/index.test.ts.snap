// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index Converts schema with optional fields 1`] = `
Object {
  "definitions": Object {
    "Error": Object {
      "$name": "Error",
      "$schema": "http://json-schema.org/draft-07/schema#",
      "properties": Object {
        "code": Object {
          "format": "int32",
          "type": "integer",
        },
        "message": Object {
          "type": "string",
        },
      },
      "required": Array [
        "code",
        "message",
      ],
    },
    "NewPet": Object {
      "$name": "NewPet",
      "$schema": "http://json-schema.org/draft-07/schema#",
      "properties": Object {
        "name": Object {
          "type": "string",
        },
        "tag": Object {
          "type": "string",
        },
      },
      "required": Array [
        "name",
      ],
    },
    "Pet": Object {
      "$name": "Pet",
      "$schema": "http://json-schema.org/draft-07/schema#",
      "additionalProperties": false,
      "properties": Object {
        "id": Object {
          "format": "int64",
          "type": "integer",
        },
        "name": Object {
          "type": "string",
        },
        "tag": Object {
          "type": "string",
        },
      },
      "required": Array [
        "name",
        "tag",
      ],
    },
  },
  "paths": Object {
    "/pets": Object {
      "get": Object {
        "200": Object {
          "items": Object {
            "$name": "Pet",
            "$schema": "http://json-schema.org/draft-07/schema#",
            "additionalProperties": false,
            "properties": Object {
              "id": Object {
                "format": "int64",
                "type": "integer",
              },
              "name": Object {
                "type": "string",
              },
              "tag": Object {
                "type": "string",
              },
            },
            "required": Array [
              "name",
              "tag",
            ],
          },
          "type": "array",
        },
        "default": Object {
          "$name": "Error",
          "$schema": "http://json-schema.org/draft-07/schema#",
          "properties": Object {
            "code": Object {
              "format": "int32",
              "type": "integer",
            },
            "message": Object {
              "type": "string",
            },
          },
          "required": Array [
            "code",
            "message",
          ],
        },
      },
      "post": Object {
        "200": Object {
          "$name": "Pet",
          "$schema": "http://json-schema.org/draft-07/schema#",
          "additionalProperties": false,
          "properties": Object {
            "id": Object {
              "format": "int64",
              "type": "integer",
            },
            "name": Object {
              "type": "string",
            },
            "tag": Object {
              "type": "string",
            },
          },
          "required": Array [
            "name",
            "tag",
          ],
        },
        "default": Object {
          "$name": "Error",
          "$schema": "http://json-schema.org/draft-07/schema#",
          "properties": Object {
            "code": Object {
              "format": "int32",
              "type": "integer",
            },
            "message": Object {
              "type": "string",
            },
          },
          "required": Array [
            "code",
            "message",
          ],
        },
      },
    },
    "/pets/{id}": Object {
      "delete": Object {
        "204": Object {},
        "default": Object {
          "$name": "Error",
          "$schema": "http://json-schema.org/draft-07/schema#",
          "properties": Object {
            "code": Object {
              "format": "int32",
              "type": "integer",
            },
            "message": Object {
              "type": "string",
            },
          },
          "required": Array [
            "code",
            "message",
          ],
        },
      },
      "get": Object {
        "200": Object {
          "$name": "Pet",
          "$schema": "http://json-schema.org/draft-07/schema#",
          "additionalProperties": false,
          "properties": Object {
            "id": Object {
              "format": "int64",
              "type": "integer",
            },
            "name": Object {
              "type": "string",
            },
            "tag": Object {
              "type": "string",
            },
          },
          "required": Array [
            "name",
            "tag",
          ],
        },
        "default": Object {
          "$name": "Error",
          "$schema": "http://json-schema.org/draft-07/schema#",
          "properties": Object {
            "code": Object {
              "format": "int32",
              "type": "integer",
            },
            "message": Object {
              "type": "string",
            },
          },
          "required": Array [
            "code",
            "message",
          ],
        },
      },
    },
  },
}
`;
